yarn-project/pxe/src/pxe_oracle_interface/pxe_oracle_interface.ts

```diff
@@ -565,60 +560,38 @@ export class PXEOracleInterface implements ExecutionDataProvider {
       }
 
-      // We filter the logs by block number and store them in the map.
-      logsMap.set(
-        recipient.toString(),
-        logsForRecipient.filter(log => log.blockNumber <= maxBlockNumber),
-      );
-
       // At this point we have processed all the logs for the recipient so we store the new largest indexes in the db.
       await this.taggingDataProvider.setTaggingSecretsIndexesAsRecipient(
         Object.entries(newLargestIndexMapToStore).map(
           ([appTaggingSecret, index]) => new IndexedTaggingSecret(Fr.fromHexString(appTaggingSecret), index),
         ),
       );
     }
-    return logsMap;
   }
 
-  /**
-   * Processes the tagged logs returned by syncTaggedLogs by decrypting them and storing them in the database.
-   * @param contractAddress - The address of the contract that the logs are tagged for.
-   * @param logs - The logs to process.
-   * @param recipient - The recipient of the logs.
-   */
-  public async processTaggedLogs(
-    contractAddress: AztecAddress,
-    logs: TxScopedL2Log[],
-    recipient: AztecAddress,
-    simulator?: AcirSimulator,
-  ): Promise<void> {
-    for (const scopedLog of logs) {
-      if (scopedLog.isFromPublic) {
-        throw new Error('Attempted to decrypt public log');
-      }
+  async #storeLogsCapsules(contractAddress: AztecAddress, recipient: AztecAddress, logs: TxScopedL2Log[]) {
+    // Build all capsules upfront with their tx effects
+    const logsCapsules = await Promise.all(
+      logs.map(async scopedLog => {
+        const txEffect = await this.aztecNode.getTxEffect(scopedLog.txHash);
+        if (!txEffect) {
+          throw new Error(`Could not find tx effect for tx hash ${scopedLog.txHash}`);
+        }
 
-      // Log processing requires the note hashes in the tx in which the note was created. We are now assuming that the
-      // note was included in the same block in which the log was delivered - note that partial notes will not work this
-      // way.
-      const txEffect = await this.aztecNode.getTxEffect(scopedLog.txHash);
-      if (!txEffect) {
-        throw new Error(`Could not find tx effect for tx hash ${scopedLog.txHash}`);
-      }
+        const logCapsule = new LogCapsule(
+          scopedLog.log.toFields(),
+          scopedLog.txHash.hash,
+          txEffect.data.noteHashes,
+          txEffect.data.nullifiers[0],
+          recipient,
+          scopedLog.logIndexInTx,
+        );
 
-      // This will trigger calls to the deliverNote oracle
-      await this.callProcessLog(
-        contractAddress,
-        scopedLog.log.toFields(),
-        scopedLog.txHash,
-        txEffect.data.noteHashes,
-        txEffect.data.nullifiers[0],
-        scopedLog.logIndexInTx,
-        recipient,
-        simulator,
-      );
-    }
-    return;
+        return logCapsule.toFields();
+      }),
+    );
+
+    return this.capsuleDataProvider.appendToCapsuleArray(contractAddress, LOG_CAPSULE_ARRAY_BASE_SLOT, logsCapsules);
```

Comment:
It'd be better if the contract passed the slot to use to PXE. That'd avoid having to have synced constants.

Reply:
Good idea. Addressed